### DISCUSSION TITLE
「分享」Android 使用 HTTP Shortcuts 录入笔记

### DISCUSSION THREAD

**Discussion by eallion** [usememos/memos#315] [Ideas]
## 下载地址：
- GitHub: https://github.com/Waboodoo/HTTP-Shortcuts/releases 
- Google Play: https://play.google.com/store/apps/details?id=ch.rmy.android.http_shortcuts&hl=en_US&gl=US

### 1. 新建变量
> 变量用于快捷方式调用，此处作用是用于输入 Memos 的文本框。 

右上角三个点
⬇︎
 选择：`变量` 
⬇︎
 右下角➕加号按钮
⬇︎
 选择：`“输入文本”` 
⬇︎
 填入：`“名称”`（随意取名，如：memos）
⬇︎
 右上角✔︎，确认

### 2. 新建快捷方式
返回 App 主页面
⬇︎ 
右下角`➕`加号按钮
⬇︎
选择：`新建快捷方式` 
⬇︎
填入：`“名称”`（随意取名，如：memos）
⬇︎
 选择：`基本设置`
　　⤵︎
　　`方法` ➡︎ 选择：`POST` 
　　`URL`&nbsp;&nbsp;➡︎ 输入：`自己的 Memos API` ：`https://memos.xx.com/api/v1/memo`
⬇︎
返回后选择：`请求体/请求参数`
　　⤵︎
　　`返回体类型`&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;➡︎ 选择：`自定义类型`
　　`Content-type`&nbsp;➡︎ 填入：`application/json`
　　`请求体`&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;➡︎ 输入：`{"content": ""}` ↪︎ 添加变量后变成：`{"content": "{memos}"}`
　　（特别注意：在这里第2个`双引号`中插入第1步新建的变量，需要手动选择，如：memos，）
　　（`请求体`里的完整内容变成这样：`{"content": "{memos}"}`。）
⬇︎
认证：
　　⤵︎
　　`认证方式` ➡︎ 选择：`令牌认证` 
　　`令牌`&nbsp;&nbsp;➡︎ 输入：`自己的 Memos Access Tokens` ：`eyJh****.eyJu****.tGBD****-2RrL****`
⬇︎
右上角的▶︎按钮可以测试，✔︎确认。

### 3. 其他
至此，基本功能已测试可用。
其他细节可自己微调。
长按此快捷方式，可以`放在主屏幕上`。

下文中有一个导出文件，选择导入就可以了。（单文本 Json 文件）
测试版本：

- memos: v0.21.0
- shortcuts: v3.11.1

需要修改 `变量` 中的 `HOST` 和 `Token` 的值。

**Comment by eallion** [comment]
APP instead https://github.com/mudkipme/MoeMemosAndroid

**Reply by toyqiu** [reply]
大佬，openid作废了，access token要怎么用呀？

**Reply by eallion** [reply]
已经修改，测试可用。

**Comment by eallion** [comment]
[memos_0.21.0_for_shortcut_3.11.1.zip](https://github.com/usememos/memos/files/14969334/memos_0.21.0_for_shortcut_3.11.1.zip)

导入文件，测试版本：

- memos: v0.21.0
- shortcuts: v3.11.1

需要修改 `变量` 中的 `HOST` 和 `Token` 的值。

**Reply by toyqiu** [reply]
> [memos_0.21.0_for_shortcut_3.11.1.zip](https://github.com/usememos/memos/files/14969334/memos_0.21.0_for_shortcut_3.11.1.zip)
> 
> 导入文件，测试版本：
> 
> * memos: v0.21.0
> * shortcuts: v3.11.1
> 
> 需要修改 `变量` 中的 `HOST` 和 `Token` 的值。

请教一下，这个只能发文字，有发图片的快捷方法嘛

**Reply by eallion** [reply]
现在已经没用这个了，需要自行摸索一下。
